scripts/check-dependencies.js

```diff
@@ -1,328 +0,0 @@
- * should install its own dependencies. We're doing this for two reasons:
- *
- *   1. There appear to be bugs in NPM that prevent us from doing `lerna link convert`
- *      and moving devDependencies to the root. We could never get this to work
- *      with packages that depend on each other within the monorepo.
- *
- *   2. When we generate a project we want to be able to make a copy of the
- *      template-retail-react-app package and give it to an end-user as a starting point.
- *      This is *much* easier to do if that package doesn't have "secret" dependencies on
- *      libraries installed at the root that aren't explicitly listed in the pwa's package.json.
- *
- * Think carefully about those choices before making a change to this setup.
- */
-const path = require('path')
-const fs = require('fs')
-const program = require('commander')
-const semver = require('semver')
-
-program.description(
-    `Check that all packages are using the same versions of crucial libraries (eg.` +
-        `compilers, linters etc) and their config files\n` +
-        `Exits with a non-zero status if there is a problem with package dependencies.`
-)
-
-program.option('--fix', 'try to fix errors by copying the right dependencies and configs packages')
-
-program.parse(process.argv)
-
-const rootDir = path.join(__dirname, '..')
-const packagesDir = path.join(rootDir, 'packages')
-const assetsDir = path.join(__dirname, 'assets')
-
-const excludedPackages = [
-    // The 'template-express-minimal' app is explicitly designed to show what
-    // you can do without our whole toolchain!
-    'template-express-minimal'
-]
-
-// We are going to copy these files into each package from this script, in order to
-// standardize them.
-const commonConfigs = ['.prettierrc.yaml']
-
-// For dependencies that we want to use the same version in all packages,
-// use this package as the source of truth for the version to use
-const COMMON_DEPS_SOURCE_OF_TRUTH = 'template-retail-react-app'
-
-// We are going to copy these devDependencies into each package from this script,
-// in order to standardize them.
-const commonDevDeps = {
-    'cross-env':
-        // eslint-disable-next-line @typescript-eslint/no-var-requires
-        require(`../packages/${COMMON_DEPS_SOURCE_OF_TRUTH}/package.json`).devDependencies['cross-env']
-}
-if (!commonDevDeps['cross-env']) {
-    throw new Error(
-        `Expected "${COMMON_DEPS_SOURCE_OF_TRUTH} to have "cross-env" as a dependency. You should revisit the assumptions in ./scripts/check-dependencies.js.`
-    )
-}
-
-const readJSON = (path) => JSON.parse(fs.readFileSync(path))
-
-const writeJSON = (path, content) => fs.writeFileSync(path, JSON.stringify(content, null, 2))
-
-const listPackages = () => {
-    return fs
-        .readdirSync(packagesDir)
-        .filter((x) => excludedPackages.indexOf(x) < 0)
-        .map((dir) => path.join(packagesDir, dir))
-}
```

Comment:
It looks to me that check-dependencies.js was concerned with only the packages in our monorepo. So it would not try to check `react` dependency version, for example. 

So I feel that replacing the script with `syncpack` is not really apple-to-apple comparison. Was there something unique that the script did that `syncpack` does not do?

Reply:
You could do that with [version groups](https://jamiemason.github.io/syncpack/guide/version-groups), to ignore any dependencies not developed in your repo. I can try and help if you have any trouble.

Reply:
@vmarta I think what you just said is exactly one of the problem with the script, it was built a while ago, maintaining the script seems like a trouble for us because we don't even know what it does and what is it trying to do. I think simplifying the setup is important for us.

Based on the comments in that script, I do believe the main purpose of it is to keep libraries use the same version which is exactly what `syncpack` does. There might be some small nuances, but i don't feel like it's important to keep them as long as we reach our goals: **make the packages use the exact same version of libraries.**

```
program.description(
    `Check that all packages are using the same versions of crucial libraries (eg.` +
        `compilers, linters etc) and their config files\n` +
        `Exits with a non-zero status if there is a problem with package dependencies.`
)
```

Reply:
@kevinxh all right, let's proceed with the changes 👍